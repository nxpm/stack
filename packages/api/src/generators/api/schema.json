{
  "$schema": "http://json-schema.org/schema",
  "cli": "nx",
  "id": "Api",
  "title": "",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What name would you like to use?"
    },
    "tags": {
      "type": "string",
      "description": "Add tags to the project (used for linting)",
      "alias": "t"
    },
    "directory": {
      "type": "string",
      "description": "A directory where the project is placed",
      "alias": "d"
    },
    "type": {
      "type": "string",
      "enum": ["nest"],
      "description": "Type or API project",
      "default": "nest"
    }
  },
  "required": ["name"]
}
