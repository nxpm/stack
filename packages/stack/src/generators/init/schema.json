{
  "$schema": "http://json-schema.org/schema",
  "cli": "nx",
  "id": "Init",
  "title": "",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What name would you like to use?"
    },
    "apiName": {
      "type": "string",
      "description": "Name of the API project",
      "default": "api"
    },
    "mobileName": {
      "type": "string",
      "description": "Name of the Mobile project",
      "default": "mobile"
    },
    "webName": {
      "type": "string",
      "description": "Name of the Web project",
      "default": "web"
    },
    "tags": {
      "type": "string",
      "description": "Add tags to the project (used for linting)",
      "alias": "t"
    },
    "directory": {
      "type": "string",
      "description": "A directory where the project is placed",
      "alias": "d"
    }
  },
  "required": ["name"]
}
