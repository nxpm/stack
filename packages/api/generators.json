{
  "$schema": "http://json-schema.org/schema",
  "name": "api",
  "version": "0.0.1",
  "generators": {
    "api": {
      "factory": "./src/generators/api/generator-api#generatorApi",
      "schema": "./src/generators/api/schema.json",
      "description": "api generator"
    },
    "api-e2e": {
      "factory": "./src/generators/api-e2e/generator-api-e2e#generatorApiE2e",
      "schema": "./src/generators/api-e2e/schema.json",
      "description": "api-e2e generator"
    },
    "api-feature": {
      "factory": "./src/generators/api-feature/generator-api-feature#generatorApiFeature",
      "schema": "./src/generators/api-feature/schema.json",
      "description": "api-feature generator"
    },
    "api-lib": {
      "factory": "./src/generators/api-lib/generator-api-lib#generatorApiLib",
      "schema": "./src/generators/api-lib/schema.json",
      "description": "api-lib generator"
    }
  }
}
