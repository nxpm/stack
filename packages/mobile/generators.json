{
  "$schema": "http://json-schema.org/schema",
  "name": "mobile",
  "version": "0.0.1",
  "generators": {
    "mobile": {
      "factory": "./src/generators/mobile/generator-mobile#generatorMobile",
      "schema": "./src/generators/mobile/schema.json",
      "description": "mobile generator"
    },
    "mobile-base": {
      "factory": "./src/generators/mobile-base/generator-mobile-base#generatorMobileBase",
      "schema": "./src/generators/mobile-base/schema.json",
      "description": "mobile-base generator"
    },
    "mobile-feature": {
      "factory": "./src/generators/mobile-feature/generator-mobile-feature#generatorMobileFeature",
      "schema": "./src/generators/mobile-feature/schema.json",
      "description": "mobile-feature generator"
    },
    "mobile-lib": {
      "factory": "./src/generators/mobile-lib/generator-mobile-lib#generatorMobileLib",
      "schema": "./src/generators/mobile-lib/schema.json",
      "description": "mobile-lib generator"
    },
    "mobile-ui": {
      "factory": "./src/generators/mobile-ui/generator-mobile-ui#generatorMobileUi",
      "schema": "./src/generators/mobile-ui/schema.json",
      "description": "mobile-ui generator"
    },
    "mobile-ui-lib": {
      "factory": "./src/generators/mobile-ui-lib/generator-mobile-ui-lib#generatorMobileUiLib",
      "schema": "./src/generators/mobile-ui-lib/schema.json",
      "description": "mobile-ui-lib generator"
    }
  }
}
