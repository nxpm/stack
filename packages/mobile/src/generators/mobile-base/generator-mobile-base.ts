import { Tree } from '@nrwl/devkit'
import { MobileBaseGeneratorSchema } from './schema'
import { generateMobileAssets } from './generate-mobile-assets'
import { generateMobileStyle } from './generate-mobile-style'
import { generateMobileEnvironments } from './generate-mobile-environments'

export async function generatorMobileBase(host: Tree, options: MobileBaseGeneratorSchema) {
  await generateMobileAssets(host, {
    ...options,
    directory: options.directory || options.mobileName,
    name: 'assets',
  })
  await generateMobileEnvironments(host, {
    ...options,
    directory: options.directory || options.mobileName,
    name: 'environments',
  })
  await generateMobileStyle(host, {
    ...options,
    directory: options.directory || options.mobileName,
    name: 'style',
  })
}
