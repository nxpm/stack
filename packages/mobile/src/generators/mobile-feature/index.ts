export * from './create-mobile-feature'
export * from './generate-mobile-feature-about'
export * from './generate-mobile-feature-account'
export * from './generate-mobile-feature-auth'
export * from './generate-mobile-feature-core'
export * from './generate-mobile-feature-dashboard'
export * from './generate-mobile-feature-layout'
export * from './generate-mobile-feature-shell'
export * from './generator-mobile-feature'
