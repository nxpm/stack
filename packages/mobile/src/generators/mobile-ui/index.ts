export * from './generate-mobile-ui-lib-button'
export * from './generate-mobile-ui-lib-form'
export * from './generate-mobile-ui-lib-icon'
export * from './generate-mobile-ui-lib-loader'
export * from './generate-mobile-ui-lib-page'
export * from './generate-mobile-ui-lib-page-header'
export * from './generate-mobile-ui-lib-sidebar-page'
export * from './generate-mobile-ui-lib-table'
