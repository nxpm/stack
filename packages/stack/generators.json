{
  "$schema": "http://json-schema.org/schema",
  "name": "stack",
  "version": "0.0.1",
  "generators": {
    "stack": {
      "factory": "./src/generators/stack/generator",
      "schema": "./src/generators/stack/schema.json",
      "description": "stack generator"
    }
  }
}
