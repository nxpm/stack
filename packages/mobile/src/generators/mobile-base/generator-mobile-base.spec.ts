import { createTreeWithEmptyWorkspace } from '@nrwl/devkit/testing'
import { readJson, readProjectConfiguration, Tree } from '@nrwl/devkit'

import { generatorMobileBase } from './generator-mobile-base'
import { MobileBaseGeneratorSchema } from './schema'
import { applicationGenerator } from '@nrwl/angular/generators'

describe('mobile-base generator', () => {
  let appTree: Tree
  const options: MobileBaseGeneratorSchema = { name: 'assets', mobileName: 'mobile-test' }

  beforeEach(() => {
    appTree = createTreeWithEmptyWorkspace()
  })

  it('should run successfully', async () => {
    await applicationGenerator(appTree, { name: options.mobileName })
    await generatorMobileBase(appTree, options)

    const nxJson = readJson(appTree, 'nx.json')
    expect(nxJson).toMatchSnapshot()

    const configApp = readProjectConfiguration(appTree, `${options.mobileName}`)
    expect(configApp).toMatchSnapshot()

    const configAssets = readProjectConfiguration(appTree, `${options.mobileName}-assets`)
    expect(configAssets).toMatchSnapshot()

    const configEnvironments = readProjectConfiguration(appTree, `${options.mobileName}-environments`)
    expect(configEnvironments).toMatchSnapshot()
  })
})
