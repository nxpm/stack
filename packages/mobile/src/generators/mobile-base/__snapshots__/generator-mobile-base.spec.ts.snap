// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mobile-base generator should run successfully 1`] = `
Object {
  "affected": Object {
    "defaultBase": "master",
  },
  "npmScope": "proj",
  "projects": Object {
    "mobile-test": Object {
      "tags": Array [],
    },
    "mobile-test-assets": Object {
      "tags": Array [],
    },
    "mobile-test-e2e": Object {
      "implicitDependencies": Array [
        "mobile-test",
      ],
      "tags": Array [],
    },
    "mobile-test-environments": Object {
      "tags": Array [],
    },
    "mobile-test-style": Object {
      "tags": Array [],
    },
  },
  "tasksRunnerOptions": Object {
    "default": Object {
      "options": Object {
        "cacheableOperations": Array [
          "build",
          "lint",
          "test",
          "e2e",
        ],
      },
      "runner": "@nrwl/workspace/tasks-runners/default",
    },
  },
}
`;

exports[`mobile-base generator should run successfully 2`] = `
Object {
  "prefix": "proj",
  "projectType": "application",
  "root": "apps/mobile-test",
  "sourceRoot": "apps/mobile-test/src",
  "tags": Array [],
  "targets": Object {
    "build": Object {
      "configurations": Object {
        "development": Object {
          "buildOptimizer": false,
          "extractLicenses": false,
          "namedChunks": true,
          "optimization": false,
          "sourceMap": true,
          "vendorChunk": true,
        },
        "production": Object {
          "budgets": Array [
            Object {
              "maximumError": "1mb",
              "maximumWarning": "500kb",
              "type": "initial",
            },
            Object {
              "maximumError": "4kb",
              "maximumWarning": "2kb",
              "type": "anyComponentStyle",
            },
          ],
          "fileReplacements": Array [
            Object {
              "replace": "libs/mobile-test/environments/src/environments/environment.ts",
              "with": "libs/mobile-test/environments/src/environments/environment.prod.ts",
            },
          ],
          "outputHashing": "all",
        },
      },
      "defaultConfiguration": "production",
      "executor": "@angular-devkit/build-angular:browser",
      "options": Object {
        "assets": Array [
          Object {
            "glob": "favicon.ico",
            "input": "libs/mobile-test/assets/src",
            "output": "./",
          },
          Object {
            "glob": "**/*",
            "input": "libs/mobile-test/assets/src/assets",
            "output": "assets",
          },
        ],
        "index": "apps/mobile-test/src/index.html",
        "main": "apps/mobile-test/src/main.ts",
        "outputPath": "dist/apps/mobile-test",
        "polyfills": "apps/mobile-test/src/polyfills.ts",
        "scripts": Array [],
        "styles": Array [
          "apps/mobile-test/src/styles.css",
        ],
        "tsConfig": "apps/mobile-test/tsconfig.app.json",
      },
      "outputs": Array [
        "{options.outputPath}",
      ],
    },
    "extract-i18n": Object {
      "executor": "@angular-devkit/build-angular:extract-i18n",
      "options": Object {
        "browserTarget": "mobile-test:build",
      },
    },
    "lint": Object {
      "executor": "@nrwl/linter:eslint",
      "options": Object {
        "lintFilePatterns": Array [
          "apps/mobile-test/src/**/*.ts",
          "apps/mobile-test/src/**/*.html",
        ],
      },
    },
    "serve": Object {
      "configurations": Object {
        "development": Object {
          "browserTarget": "mobile-test:build:development",
        },
        "production": Object {
          "browserTarget": "mobile-test:build:production",
        },
      },
      "defaultConfiguration": "development",
      "executor": "@angular-devkit/build-angular:dev-server",
    },
    "test": Object {
      "executor": "@nrwl/jest:jest",
      "options": Object {
        "jestConfig": "apps/mobile-test/jest.config.js",
        "passWithNoTests": true,
      },
      "outputs": Array [
        "coverage/apps/mobile-test",
      ],
    },
  },
}
`;

exports[`mobile-base generator should run successfully 3`] = `
Object {
  "prefix": "proj",
  "projectType": "library",
  "root": "libs/mobile-test/assets",
  "sourceRoot": "libs/mobile-test/assets/src",
  "tags": Array [],
  "targets": Object {},
}
`;

exports[`mobile-base generator should run successfully 4`] = `
Object {
  "prefix": "proj",
  "projectType": "library",
  "root": "libs/mobile-test/environments",
  "sourceRoot": "libs/mobile-test/environments/src",
  "tags": Array [],
  "targets": Object {
    "lint": Object {
      "executor": "@nrwl/linter:eslint",
      "options": Object {
        "lintFilePatterns": Array [
          "libs/mobile-test/environments/src/**/*.ts",
          "libs/mobile-test/environments/src/**/*.html",
        ],
      },
    },
    "test": Object {
      "executor": "@nrwl/jest:jest",
      "options": Object {
        "jestConfig": "libs/mobile-test/environments/jest.config.js",
        "passWithNoTests": true,
      },
      "outputs": Array [
        "coverage/libs/mobile-test/environments",
      ],
    },
  },
}
`;
